decomp/semantics/uds.py

```diff
@@ -154,98 +230,209 @@ def from_conll(cls,
+            or normalized form and whose values are paths to JSON files containing
+            document-level annotations
         name
             corpus name to be appended to the beginning of graph ids
         """
         predpatt_corpus = PredPattCorpus.from_conll(corpus, name=name)
-        predpatt_graphs = {name: UDSGraph(g, name)
+        predpatt_sentence_graphs = {name: UDSSentenceGraph(g, name)
                            for name, g in predpatt_corpus.items()}
+        predpatt_documents = cls._initialize_documents(predpatt_sentence_graphs)
 
-        # Although neither RawUDSDataset nor NormalizedUDSDataset currently
-        # modifies the from_json method of the abstract base class, we don't
-        # want to assume this, and users should be deliberate in specifying
-        # the annotation type.
-        processed_annotations = []
-        for ann_type, ann_path in annotations.items():
+        # Process sentence-level graph annotations
+        processed_sentence_annotations = []
+        for ann_type, ann_path in sentence_annotations.items():
             if ann_type == 'raw':
                 for anno in ann_path:
-                    processed_annotations.append(RawUDSDataset.from_json(anno))
+                    processed_sentence_annotations.append(RawUDSDataset.from_json(anno))
             elif ann_type == 'normalized':
                 for anno in ann_path:
-                    processed_annotations.append(NormalizedUDSDataset.from_json(anno))
+                    processed_sentence_annotations.append(NormalizedUDSDataset.from_json(anno))
             else:
                 raise ValueError('Unrecognized annotation type {0} '\
-                                 'for annotation {1}.'.format(ann_type, ann_path))
+                                 'for sentence annotation {1}.'.format(ann_type, ann_path))
 
-        return cls(predpatt_graphs, processed_annotations)
+        # Process document-level graph annotations
+        processed_document_annotations = []
+        for ann_type, ann_path in document_annotations.items():
+            if ann_type == 'raw':
+                for anno in ann_path:
+                    processed_document_annotations.append(RawUDSDataset.from_json(anno))
+            elif ann_type == 'normalized':
+                for anno in ann_path:
+                    processed_document_annotations.append(NormalizedUDSDataset.from_json(anno))
+            else:
+                raise ValueError('Unrecognized annotation type {0} '\
+                                 'for document annotation {1}.'.format(ann_type, ann_path))
+
+        return cls(predpatt_sentence_graphs, predpatt_documents, 
+                   processed_sentence_annotations, processed_document_annotations)
+
+    @classmethod
+    def _load_ud_ids(cls, sentence_ids_only: bool = False) -> Dict[str, Dict[str, str]]:
+
+        # The graph dictionary is sub-keyed on root syntax node,
```

Comment:
We don't need to track the root node ID in the JSON, correct? Should we just flatten the structure of that JSON?

Reply:
Yeah, we can do that.